lib/state/tradeslip/tradeslipItemsState.ts

```diff
@@ -0,0 +1,371 @@
+
+  const poolZtgBalances = usePoolZtgBalance(pools ?? []);
+
+  const traderAssets = useAccountAssetBalances(
+    items.map((item) => ({
+      account: signer?.address,
+      assetId: item.assetId,
+    })),
+  );
+
+  const poolAssetBalances = useAccountAssetBalances(
+    items.map((item) => ({
+      account: pools?.find((p) => p.marketId == getMarketIdOf(item.assetId))
+        ?.accountId,
+      assetId: item.assetId,
+    })),
+  );
+
+  const balancesKey = {
+    traderZtgBalance: traderZtgBalance.toString(),
+    traderAssets: traderAssets?.map((a) => a?.toString()),
+  };
+
+  const query = useQueries({
+    queries: items.map((item) => {
+      const pool = pools?.find(
+        (p) => p.marketId == getMarketIdOf(item.assetId),
+      );
+
+      const amount = new Decimal(item.amount).mul(ZTG);
+      const assetIndex = getIndexOf(item.assetId);
+      const saturatedData = saturatedIndex?.[pool?.poolId];
+      const asset = saturatedData?.assets[assetIndex];
+      const market = saturatedData?.market;
+
+      const ztgWeight = pool
+        ? getAssetWeight(pool, { Ztg: null }).unwrap()
+        : null;
+
+      const assetWeight = asset
+        ? getAssetWeight(pool, asset?.assetId).unwrap()
+        : null;
+
+      const swapFee = pool?.swapFee
+        ? new Decimal(pool.swapFee)
+        : new Decimal(0);
+
+      const poolZtgBalance =
+        !poolZtgBalances[0] || isNA(poolZtgBalances[0])
+          ? null
+          : new Decimal(poolZtgBalances[0].balance.data.free.toString());
+
+      const traderAssetBalance =
+        !traderAssets?.[0] || isNA(traderAssets?.[0])
+          ? na("Account balance not available.")
+          : new Decimal(traderAssets?.[0].free.toString());
+
+      const poolAssetBalance =
+        !poolAssetBalances[0] || isNA(poolAssetBalances[0])
+          ? null
+          : new Decimal(poolAssetBalances[0]?.free.toString());
+
+      const tradeablePoolBalance =
+        !poolAssetBalance || isNA(poolAssetBalance)
+          ? null
+          : new Decimal(poolAssetBalance).mul(MAX_IN_OUT_RATIO);
+
+      const enabled = Boolean(
+        item &&
+          pool &&
+          asset &&
+          market &&
+          ztgWeight &&
+          assetWeight &&
+          swapFee &&
+          poolZtgBalance &&
+          traderAssetBalance &&
+          poolAssetBalance &&
+          tradeablePoolBalance,
+      );
+
+      return {
+        enabled: Boolean(sdk) && isRpcSdk(sdk) && enabled,
+        keepPreviousData: true,
+        queryKey: [
+          id,
+          rootKey,
+          itemKey(item),
+          amount,
+          signer?.address,
+          balancesKey,
+        ],
+        queryFn: async () => {
+          if (!enabled || !sdk || !isRpcSdk(sdk)) {
+            return null;
+          }
+
+          const max = (() => {
+            if (item.action === "buy") {
+              const maxTokens = isNA(traderZtgBalance)
+                ? new Decimal(Infinity)
```

Comment:
Setting it to 0 means the input cant be changed for users that havent connected their wallet, meaning we dont know their ZTG balance.. But Im changing it to the `tradeablePoolBalance` so that the user can set the input to the maximum allowed to buy from the pool granted they have the ZTG.